### ISSUE TITLE
Master Branch Rename: API

### ISSUE THREAD

**Issue by coreycarvalho** [department-of-veterans-affairs/notification-api#1905]
## User Story - Business Need

- [x] Ticket is understood, and QA has been contacted (if the ticket has a QA label).

### User Story(ies)

**As a** DevOps engineer    
**I want** to rename the `master` branch to `main` in this repository  
**So that** our branch naming aligns with industry standards and promotes inclusive terminology.

### Additional Info and Resources
- [GitHub guide on renaming the default branch](https://docs.github.com/en/github/administering-a-repository/renaming-a-branch)
- Steps to rename the branch:
  1. Update local clones:
     ```sh
     git branch -m master main
     git fetch origin
     git branch -u origin/main main
     git remote set-head origin -a
     ```
  2. Update any Github Actions to use the `main` branch.
  3. Update documentation to reflect the branch name change.

## Acceptance Criteria
- [x] The `master` branch has been successfully renamed to `main` in this repository.
- [x] Github Actions have been updated to reference the `main` branch.
- [x] All documentation and code references to the `master` branch have been updated to `main`.
- [x] [Team is notified in Slack to run the `Update local clones` directions from above.](https://oddball.slack.com/archives/C036AE08HA7/p1724258583877339?thread_ts=1724258176.910019&cid=C036AE08HA7)

## QA Considerations
- Ensure that all automated tests run successfully after the branch rename.
- Verify that the deployment pipeline still functions correctly with the new branch name.
- Check that any links or references to the branch in the documentation are updated and correct.

## Potential Dependencies
- Coordination with other teams (IIR, VEText) that might have dependencies on the `master` branch.

**Comment by cris-oddball** [comment]
_CI/CD pipeline has been updated to reference the main branch._

Please reword this to note any and all actions. We have far more than just what is in the pipeline.

**Comment by kbelikova-oddball** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/VA-Notify-620d21369d810a00146ed9c8/issues/department-of-veterans-affairs/notification-api/1905?planning-poker) with Zenhub @coreycarvalho @cris-oddball @EvanParish @k-macmillan @kalbfled @MackHalliday @mchlwellman

**Comment by npmartin-oddball** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/VA-Notify-620d21369d810a00146ed9c8/issues/department-of-veterans-affairs/notification-api/1905?planning-poker) with Zenhub @coreycarvalho @cris-oddball @EvanParish @k-macmillan @kalbfled @MackHalliday @mchlwellman

**Comment by MackHalliday** [comment]
TODO
- Do all the updates locally, then change the branch
- DO NO change regression QA

**Comment by MackHalliday** [comment]
Yesterday 
- Talked to QA 
- Changed all names from master to main locally in Github Actions and Docs

Today 
- Figure out how to GRACEFULLY implement change...

**Comment by cris-oddball** [comment]
**QA TESTING IN PROGRESS**

- [ ] Dependency Update ticket includes Main, not Master (open a draft but don't save it)
- [ ] PR template includes Main (open a draft but don't save it)
- [ ] Pipeline GHA functions as expected
- [ ] Docker build GHA functions as expected (tested in Pipeline)
- [ ] Deploy lambdas GHA functions as expected (tested in Pipeline)
- [ ] Lambda functions GHA functions as expected (tested in Deploy lambdas)
- [ ] Deploy to ENV GHA functions as expected - deploy to dev
- [ ] sms load test GHA functions as expected

- [ ] Tests GHA functions as expected - need to open a PR
- N/A test GitHub Actions GHA - this is not used, not testing
- N/A Update DD Image GHA - used by the test GHA, hasn't been used in 5 months, not testing
- N/A email load test GHA - this has not worked in a long time, not testing
- N/A Deploy Release GHA  - this is now unused and superceded by the pipeline
